src/main/scala/beam/agentsim/agents/BeamAgent.scala

```diff
@@ -73,8 +73,12 @@ trait BeamAgent[T <: BeamAgentData] extends LoggingFSM[BeamAgentState, BeamAgent
   protected var _currentTick: Option[Double] = None
 
   private val chainedStateFunctions = new mutable.HashMap[BeamAgentState, mutable.Set[StateFunction]] with mutable.MultiMap[BeamAgentState,StateFunction]
-  final def chainedWhen(stateName: BeamAgentState)(stateFunction: StateFunction): Unit =
+  final def chainedWhen(stateName: BeamAgentState)(stateFunction: StateFunction): Unit = {
     chainedStateFunctions.addBinding(stateName,stateFunction)
+    when(stateName, null) { case event =>
+      handleEvent(stateName, event)
+    }
```

Comment:
Please revert this addition. The point of the _chainedWhen_ is as a replacement to _when_, not a wrapper for it. If multiple partials are registered via _when_ to the same state, only one the matching cases will run. _chainedWhen_ is designed so that all matching cases in the registered partials will run. 

If there are new code blocks not running when you expected them, you may be missing the boilerplate in the class or trait that handles events for a particular state, like this:

```
when(Initialized){
    case ev @ Event(_,_) =>
      handleEvent(stateName, ev)
  }
```